gef-ui/server/controllers/unissued-facilities/validation.js

```diff
@@ -395,6 +394,23 @@ const facilityValidation = async ({ body, query, params, facility, userToken })
     }
   }
 
+  /**
+   * @type{boolean | undefined}
+   */
+  let isUsingFacilityEndDate;
+  if (isFacilityEndDateEnabledOnGefVersion(parseDealVersion(application.version))) {
+    try {
+      isUsingFacilityEndDate = zBooleanStrictCoerce.parse(isUsingFacilityEndDateString);
```

Comment:
oooh what does `zBooleanStrictCoerce` do?

Reply:
This was introduced initially to parse env FFs (see libs\common\src\helpers\is-feature-flag-enabled.ts) - it takes any reasonable string/number/boolean and parses it to a boolean, primarily to get around the issue of `false` being truthy. Zod always throws an error if it doesn't work, hence the try/catch

Reply:
Good to know! Thanks!